oauth2/src/test/groovy/com/okta/spring/boot/oauth/OktaOAuth2AutoConfigRestTemplateTest.groovy

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.spring.boot.oauth
+
+import com.okta.spring.boot.oauth.config.OktaOAuth2Properties
+import org.mockserver.integration.ClientAndServer
+import org.mockserver.model.Cookie
+import org.mockserver.model.Header
+import org.mockserver.model.HttpRequest
+import org.mockserver.model.HttpResponse
+import org.mockserver.verify.VerificationTimes
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.HttpMethod
+import org.springframework.http.ResponseEntity
+import org.springframework.security.oauth2.core.endpoint.OAuth2AccessTokenResponse
+import org.springframework.web.client.RestTemplate
+import org.testng.annotations.AfterClass
+import org.testng.annotations.Test
+
+import static java.util.Collections.singletonList
+import static java.util.Collections.singletonMap
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.is
+
+class OktaOAuth2AutoConfigRestTemplateTest {
+    private static final String LOCALHOST = "localhost"
+    private static final int PORT = 7000
```

Comment:
Don't use a static port for testing, use a random free port, this mockserver _should_ support this out of the box, but if not, just lookup a free port first

Reply:
👍